packages/utils/esbuild.js

```diff
@@ -0,0 +1,24 @@
+import { build } from 'esbuild';
+
+// Building for CommonJS
+build({
+    entryPoints: ['src/index.ts'], // Your entry file
+    bundle: true,
+    outfile: 'build/index.cjs.js', // Output file for CommonJS
+    format: 'cjs', // Specify CommonJS format
+    platform: 'node', // Platform target
+    sourcemap: true, // Enable source maps (optional)
+    target: ['node18'], // Target environment (example: Node 12)
+    // additional options...
+}).catch(() => process.exit(1));
+
+// Building for ECMAScript Module
+build({
+    entryPoints: ['src/index.ts'], // Your entry file
+    bundle: true,
```

Comment:
Since the problems we're seeing relating to ESM are actually only visible in CJS environments (that I know of), it doesn't make sense to bundle dependencies at this step.

What do you think?

```suggestion
    bundle: false,
```